### PR TITLE
test: remove explicit any types from test files - Part 8

### PR DESCRIPTION
**Pull request by Myestery**
## Summary
Part 8 of the systematic removal of explicit `any` types from test files. Focus on Phase 2 "quick wins" - files with 1-2 instances.

## Changes
**Phase 2 - Quick Wins (9 files, 23 any types removed)**

### Component Test Helpers
- **FormRadioGroup.test.ts** (1) - ComponentProps pattern
- **UrlInput.test.ts** (1) - ComponentProps pattern + refactored to check DOM instead of internal state
- **UserAvatar.test.ts** (1) - ComponentProps pattern  
- **SettingItem.test.ts** (1) - ComponentProps pattern with flexible props
- **ApiKeyForm.test.ts** (1) - ComponentProps pattern

### Mock Typing
- **BaseTerminal.test.ts** (1) - Proper `Mock` type from vitest
- **UsageLogsTable.test.ts** (1) - Defined MockEvent interface

### Composable Tests
- **useSelectedLiteGraphItems.test.ts** (1) - Typed mockCanvas, fixed MockNode
- **useSelectionState.test.ts** (14) - Pinia store mocks with proper typing

## Patterns Established
1. **ComponentProps pattern**: `import type { ComponentProps } from 'vue-component-type-helpers'`
2. **Flexible test helpers**: `Record<string, unknown>` with `as unknown as ComponentProps<...>` for edge cases
3. **Vitest Mock typing**: `import type { Mock } from 'vitest'`
4. **Pinia store mocks**: `as unknown as ReturnType<typeof store>`

## Progress
- **This PR**: 23 any types removed across 9 files
- **Part 8 Total**: 74 any types removed across 18 files (including Phase 1)
- **All tests passing**: ✅
- **0 typecheck errors**: ✅

## Related
- Continues work from Parts 1-7
- Part of ongoing effort to eliminate all explicit `any` types from the codebase

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-8197-test-remove-explicit-any-types-from-test-files-Part-8-2ef6d73d365081a4a735d2eac55d6139) by [Unito](https://www.unito.io)

### CODE REVIEW

File: src/components/bottomPanel/tabs/terminal/BaseTerminal.test.ts [hunk 1]
```diff
@@ -151,8 +152,8 @@ describe('BaseTerminal', () => {
     // Trigger the selection change callback that was registered during mount
     expect(mockTerminal.onSelectionChange).toHaveBeenCalled()
     // Access the mock calls - TypeScript can't infer the mock structure dynamically
-    const selectionCallback = (mockTerminal.onSelectionChange as any).mock
-      .calls[0][0]
+    const mockCalls = (mockTerminal.onSelectionChange as Mock).mock.calls
```
Comment (DrJKL) [inline src/components/bottomPanel/tabs/terminal/BaseTerminal.test.ts:155]: Could you use [vi.mocked](https://vitest.dev/api/vi.html#vi-mocked)?

Comment (DrJKL) [reply src/components/bottomPanel/tabs/terminal/BaseTerminal.test.ts:155]: In general, I would avoid type assertions.

File: src/components/common/FormRadioGroup.test.ts [hunk 2]
```diff
@@ -14,13 +16,13 @@ describe('FormRadioGroup', () => {
...
     return mount(FormRadioGroup, {
       global: {
         plugins: [PrimeVue],
         components: { RadioButton }
       },
-      props,
+      props: props as unknown as ComponentProps<typeof FormRadioGroup>,
       ...options
     })
   }
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply src/components/common/FormRadioGroup.test.ts:28]: @coderabbitai Don't say that this is acceptable.

**Comment by Myestery** [comment]
Will split this into 3 or 4 parts for easy review.

**Comment by Myestery** [comment]
> Will split this into 3 or 4 parts for easy review.

also, many `as unknown as` need to be refactored before merging
